app/screens/global_threads/global_threads.tsx

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useMemo, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {Keyboard, StyleSheet, View} from 'react-native';
+import {Edge, SafeAreaView, useSafeAreaInsets} from 'react-native-safe-area-context';
+
+import NavigationHeader from '@components/navigation_header';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import {useAppState, useIsTablet} from '@hooks/device';
+import {useDefaultHeaderHeight} from '@hooks/header';
+import {popTopScreen} from '@screens/navigation';
+
+import ThreadsList from './threads_list';
+
+import type {Tab} from './threads_list/threads_list';
+
+type Props = {
+    componentId?: string;
+    currentTeamId: string;
+};
+
+const edges: Edge[] = ['left', 'right'];
+
+const styles = StyleSheet.create({
+    flex: {
+        flex: 1,
+    },
+});
+
+const GlobalThreads = ({componentId, currentTeamId}: Props) => {
+    const appState = useAppState();
+    const intl = useIntl();
+    const insets = useSafeAreaInsets();
+    const isTablet = useIsTablet();
+    const serverUrl = useServerUrl();
+
+    const theme = useTheme();
+    const defaultHeight = useDefaultHeaderHeight();
+
+    const [tab, setTab] = useState<Tab>('all');
+
+    const containerStyle = useMemo(() => {
+        const marginTop = defaultHeight + insets.top;
+        return [styles.flex, {marginTop}];
```

Comment:
Unless we are planning to change the style in the near future, I think it makes more sense to keep it all together here.
i.e. `return {flex: 1, marginTop}`

That way ReactNative doesn't need to flatten the style down the road.